vortex-array/src/encoding/mod.rs

```diff
@@ -67,6 +69,19 @@ pub trait Encoding: 'static {
     type Metadata: ArrayMetadata;
 }
 
+pub fn downcast_array<E: Encoding>(array: &ArrayData) -> VortexResult<(&E::Array, &E)>
```

Comment:
Can this be called `downcast_array_ref`? Since we have a non-ref'd version of this too.

Also maybe this should happen in a FLUP PR that converts everywhere.